### PR TITLE
feat: emit fallback configuration events

### PR DESCRIPTION
**Pull request by czeslavo**
**What this PR does / why we need it**:

Starts emitting new Kubernetes events related to the `FallbackConfiguration` feature:

- `FallbackKongConfigurationSucceeded` - emitted on successful fallback configuration push (attached to Pod)
- `FallbackKongConfigurationTranslationFailed` - emitted when a fallback cache causes translation failures in the translator (attached to causing objects)
- `FallbackKongConfigurationApplyFailed` - emitted when a fallback configuration is rejected by Kong (attached to Pod and broken objects if any)

**Which issue this PR fixes**:

Closes https://github.com/Kong/kubernetes-ingress-controller/issues/6081.

**Special notes for your reviewer**:

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [x] the `CHANGELOG.md` release notes have been updated to reflect any significant (and particularly user-facing) changes introduced by this PR

### CODE REVIEW

File: internal/dataplane/kong_client.go [hunk 1]
```diff
@@ -628,7 +640,10 @@ func extractBrokenObjectsFromUpdateError(err error) ([]fallback.ObjectHash, erro
 // sendOutToGatewayClients will generate deck content (config) from the provided kong state
 // and send it out to each of the configured gateway clients.
 func (c *KongClient) sendOutToGatewayClients(
-	ctx context.Context, s *kongstate.KongState, config sendconfig.Config,
+	ctx context.Context,
+	s *kongstate.KongState,
+	config sendconfig.Config,
+	isFallback bool,
```
Comment (programmer04) [inline internal/dataplane/kong_client.go:646]: It's a little hard to follow with such an API at least for me. Maybe having two methods
- `sendOutToGatewayClients(ctx context.Context,	s *kongstate.KongState,	config sendconfig.Config)`
- `sendOutToGatewayClientsFallback(ctx context.Context,	s *kongstate.KongState,	config sendconfig.Config)`

and inside calling the above with `isFallback = false/true`. WDYT?

File: internal/dataplane/kong_client_test.go [hunk 2]
```diff
@@ -653,6 +671,183 @@ func TestKongClient_ApplyConfigurationEvents(t *testing.T) {
...
+			updateStrategyResolver := newMockUpdateStrategyResolver(t)
+			configBuilder := newMockKongConfigBuilder()
+			eventRecorder := mocks.NewEventRecorder()
+			lastValidConfigFetcher := &mockKongLastValidConfigFetcher{}
+			kongClient := setupTestKongClient(t, updateStrategyResolver, clientsProvider, configChangeDetector, configBuilder, eventRecorder, lastValidConfigFetcher)
+			kongClient.kongConfig.FallbackConfiguration = tc.fallbackConfiguration
+
+			if tc.translationFailures {
+				configBuilder.translationFailuresToReturn = []failures.ResourceFailure{
+					lo.Must(failures.NewResourceFailure("some reason", testIngress)),
```
Comment (programmer04) [inline internal/dataplane/kong_client_test.go:801]:
```suggestion
					require.No(tc, failures.NewResourceFailure("some reason", testIngress)),
```
why not like that? But TBH it doesn't matter here is basically equivalent

Comment (czeslavo) [reply internal/dataplane/kong_client_test.go:801]: I'm not aware of any `testify` helper that would be equivalent to `lo.Must` (it panics on err, and returns a value if no error is there). I think it's fine to use such a helper in tests if you're initializing some values with static input.

Comment (programmer04) [reply internal/dataplane/kong_client_test.go:801]: yup, my bad I overlooked returned value. It's good

File: internal/dataplane/kong_client.go (hunk 1 above)
Comment (czeslavo) [reply internal/dataplane/kong_client.go:646]: Yeah, I don't like this API either, but we have to pass this flag up from `KongClient.Update` and `tryRecoveringWithFallbackConfiguration` down to `sendToClient` through `sendOutToGatewayClients` and `maybeSendOutToKonnectClient`. If we'd like to have a dedicated method for each, we'd end up with:

- `sendToClientFallback`
- `sendToClientRegular`
- `sendToClient` (implementation with fallback flag param)
- `sendOutToGatewaysClientsFallback`
- `sendOutToGatewaysClientsRegular`
- `sendOutToGatewaysClients` (implementation with fallback flag param)
- `maybeSendOutToKonnectClientFallback`
- maybeSendOutToKonnectClientRegular`
- `maybeSendOutToKonnectClient` (implementation with fallback flag param)

which seems to me like an overkill.

I considered below options:
- passing the fallback flag in the context (it would be quite implicit and easy to miss)
- extending `KongState` with `Meta` struct that would contain the `IsFallback` flag (it breaks the `KongState` - currently it only contains Kong entities and I think it's better to leave it this way)
- passing `ConfigPushMetadata` with the `IsFallback` field instead of the raw flag (this seemed to me like overkill, but maybe this would be cleaner as we could have some godocs explaining the meaning of it). WDYT?

Comment (programmer04) [reply internal/dataplane/kong_client.go:646]: I see the cardinality rise. Let's leave it for now as it is for now

**Review by programmer04 — APPROVED** [review]
